src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -84,18 +88,26 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
                 }
             }
 
+            if (refKind != RefKind.None)
+            {
+                refCustomModifiers = customModifiers.ToImmutableAndFree();
+            }
+            else
+            {
+                returnType = returnType.WithModifiers(customModifiers.ToImmutableAndFree());
+            }
+
             return new FunctionPointerMethodSymbol(
                 callingConvention,
                 refKind,
                 returnType,
-                refReadonlyModifiers,
+                refCustomModifiers,
                 syntax,
                 typeBinder,
                 diagnostics,
                 suppressUseSiteDiagnostics);
 
-            // PROTOTYPE(func-ptr): Do this right. Just making compatible changes for now
-            static CallingConvention getCallingConvention(FunctionPointerCallingConventionSyntax? callingConventionSyntax, DiagnosticBag diagnostics)
+            static CallingConvention getCallingConvention(Binder typeBinder, FunctionPointerCallingConventionSyntax? callingConventionSyntax, ArrayBuilder<CustomModifier> customModifiers, DiagnosticBag diagnostics)
```

Comment:
>typeBinder [](start = 65, length = 10)

It looks like what we need is the compilation. Having dependency on the binder feels strange, there is nothing to "bind" in any particular scope or context here.